File: ulyp-ui/src/main/java/com/ulyp/ui/renderers/RenderedMap.java
package com.ulyp.ui.renderers;

import com.ulyp.core.printers.MapRepresentation;
import com.ulyp.ui.RenderSettings;
import com.ulyp.ui.util.StyledText;
import javafx.scene.Node;

import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;

import static com.ulyp.ui.util.CssClass.*;

public class RenderedMap extends RenderedObject {

    protected RenderedMap(MapRepresentation representation, RenderSettings renderSettings) {
        super(representation.getType());

        List<RenderedObject> entries = representation.getEntries()
                .stream()
                .map(repr -> RenderedObject.of(repr, renderSettings))
                .collect(Collectors.toList());

        List<Node> texts = new ArrayList<>();

        if (renderSettings.showTypes()) {
            texts.add(StyledText.of(representation.getType().getName(), CALL_TREE_TYPE_NAME));
            texts.add(StyledText.of(": ", CALL_TREE_PLAIN_TEXT));
        }

        texts.add(StyledText.of("{", CALL_TREE_COLLECTION_BRACE));

        for (int i = 0; i < entries.size(); i++) {
            texts.add(entries.get(i));
            if (i != entries.size() - 1 || entries.size() < representation.getSize()) {
                texts.add(StyledText.of(", ", CALL_TREE_PLAIN_TEXT));
            }
        }

        if (entries.size() < representation.getSize()) {
            texts.add(StyledText.of((representation.getSize() - entries.size()) + " more...", CALL_TREE_PLAIN_TEXT));
        }

        texts.add(StyledText.of("}", CALL_TREE_COLLECTION_BRACE));

        super.getChildren().addAll(texts);
    }
}

File: ulyp-ui/src/main/kotlin/com/ulyp/ui/renderers/RenderedMap.kt
package com.ulyp.ui.renderers

import com.ulyp.core.printers.MapEntryRecord
import com.ulyp.core.printers.MapRecord
import com.ulyp.ui.RenderSettings
import com.ulyp.ui.util.CssClass
import com.ulyp.ui.util.StyledText.of
import javafx.scene.Node
import java.util.stream.Collectors

class RenderedMap(representation: MapRecord, renderSettings: RenderSettings) : RenderedObject(representation.type) {

    init {
        val entries = representation.entries
            .stream()
            .map { repr: MapEntryRecord -> of(repr, renderSettings) }
            .collect(Collectors.toList())
        val texts: MutableList<Node> = ArrayList()
        if (renderSettings.showTypes()) {
            texts.add(of(representation.type.name, CssClass.CALL_TREE_TYPE_NAME))
            texts.add(of(": ", CssClass.CALL_TREE_NODE_SEPARATOR))
        }
        texts.add(of("{", CssClass.CALL_TREE_COLLECTION_BRACKET))
        for (i in entries.indices) {
            texts.add(entries[i])
            if (i != entries.size - 1 || entries.size < representation.size) {
                texts.add(of(", ", CssClass.CALL_TREE_NODE_SEPARATOR))
            }
        }
        if (entries.size < representation.size) {
            texts.add(of((representation.size - entries.size).toString() + " more...", CssClass.CALL_TREE_NODE_SEPARATOR))
        }
        texts.add(of("}", CssClass.CALL_TREE_COLLECTION_BRACKET))
        super.getChildren().addAll(texts)
    }
}